custom_components/tuya_local/devices/plantsio_ivy_smart_planter.yaml

```diff
@@ -0,0 +1,431 @@
+# Datapoints documented at:
+# https://gist.github.com/thewade/ef9f6014f13932bd2e77d43331e2027d
+
+name: Smart Planter
+products:
+  - id: 8n3q0y4cwov8ifxt
+    name: PlantsIO Ivy Smart Planter
+primary_entity:
+  entity: sensor
+  name: Water level
+  dps:
+    - id: 108
+      type: integer
+      name: sensor
+      unit: mL
+      class: measurement
+      mapping:
+        - dps_val: 0
+          icon: "mdi:cup-outline"
+        - icon: "mdi:cup"
+    - id: 107
+      type: string
+      name: version
+secondary_entities:
+  # Sensor entities
+  - entity: sensor
+    name: Ambient light
+    class: illuminance
+    dps:
+      - id: 103
+        type: integer
+        name: sensor
+        unit: lx
+        class: measurement
+  - entity: sensor
+    name: Temperature
+    class: temperature
+    dps:
+      - id: 104
+        type: integer
+        name: sensor
+        unit: C
+        class: measurement
+        mapping:
+          - scale: 10
+  - entity: sensor
+    name: Humidity
+    class: humidity
+    dps:
+      - id: 105
+        type: integer
+        name: sensor
+        unit: "%"
+        class: measurement
+  - entity: sensor
+    name: Soil moisture
+    class: moisture
+    dps:
+      - id: 122
+        name: sensor
+        type: integer
+        optional: true
+        unit: "%"
+  - entity: sensor
+    name: Water status
+    icon: "mdi:watering-can"
+    class: enum
+    dps:
+      - id: 106
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: "Initializing"
+          - dps_val: 2
+            value: "Drinking"
+          - value: "Unknown state"
+  - entity: sensor
+    name: Light status
+    icon: "mdi:brightness-5"
+    class: enum
+    dps:
+      - id: 138
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: "Acceptable"
+          - dps_val: 2
+            value: "Good"
+          - dps_val: 3
+            value: "Exceptional"
+          - dps_val: 4
+            value: "Too Much"
+          - dps_val: 5
+            value: "Insufficient"
+          - value: "Unknown state"
+  - entity: sensor
+    name: Temperature status
+    icon: "mdi:thermometer"
+    class: enum
+    dps:
+      - id: 139
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: "Good"
+          - dps_val: 2
+            value: "Hot"
+            icon: "mdi:thermometer-high"
+          - dps_val: 3
+            value: "Cold"
+            icon: "mdi:thermometer-low"
+          - value: "Unknown state"
+  - entity: sensor
+    name: Humidity status
+    class: enum
```

Comment:
Sensors with only two states could maybe be made into binary_sensor instead, especially if there is a relevant class defined for binary_sensor.

Reply:
I would expect there to be multiple states from the "AI" derived sensors. 

e.g. Good, Dry, Too humid, Too dry

However I have no been able to determine all the values in my environment, and as I see with Light status enum the values are not in order so I don't want to guess.

The developers seems to had expressed interested in home assistant status, so I have emailed them to see if they are willing to share the mapping.